clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -250,6 +251,7 @@ public void run() {
                 run(time.milliseconds());
             } catch (Exception e) {
                 log.error("Uncaught error in kafka producer I/O thread: ", e);
+                e.printStackTrace();
```

Comment:
have we found out why log.error doesn't print out stack trace?

Reply:
No, probably some log4j settings. Since we are moving to 2.4 and this version will be only used as a release candidate for BMM, we might just want to print this temporarily and revert this later.